token_dispenser/token_dispenser_lambda.py

```diff
@@ -0,0 +1,175 @@
+    :return: a json object containing the token data.
+    """
+    global cached_cert_file
+    logger = shared_logger()
+    try:
+        if cached_cert_file is not None and os.path.exists(cached_cert_file.name) and os.path.getsize(cached_cert_file.name) >0:
+            logger.debug(f"found cached cert file {cached_cert_file.name}")
+            get_token(url=config.LAUNCHPAD_GETTOKEN_URL, cert_file=cached_cert_file.name)
+        else:
+            logger.debug(f"cached cert file not found")
+            p12_file = download_s3_file(bucket_name=config.LAUNCHPAD_PFX_FILE_S3_BUCKET,
+                                                key=config.LAUNCHPAD_PFX_FILE_S3_KEY, local_storage_dir='/tmp')
+            logger.info(f"p12 file downloaded from s3 successfully to: {p12_file}")
+            password = get_secret_value(config.LAUNCHPAD_PFX_PASSWORD_SECRET_ARN)
+            private_key, cert, additional_certs = decode_pkcs12(p12_file, password)
+            cached_cert_file = build_cached_cert_file(private_key, cert)
+            # Create launchpad token
+            token_json = get_token(url=config.LAUNCHPAD_GETTOKEN_URL, cert_file=cached_cert_file.name)
+        put_token(client_id, json.dumps(token_json), int(token_json['expires_at']))
+        return token_json
+    except Exception as e:
+        print(f"Failed on get_new_token process: {e}")
+        raise e
+
+
+def satisfy_minimum_alive_secs(expires_at:int, minimum_alive_secs:int) -> bool:
+    # if the minimum_alive_sec is not provided by requester or value less than 0
+    # it means the request does not care so we will assume whatever cached is ok
+    if minimum_alive_secs is None or minimum_alive_secs <= 0:
+        return True
+    if expires_at - time.time() > minimum_alive_secs:
+        return True
+    else:
+        return False
+
+def is_client_id_valid(client_id:str) -> bool:
+    pattern = re.compile(r'^[a-zA-Z0-9]{3,32}$')
+    return bool(pattern.match(client_id))
+
+
+def is_minimum_alive_secs_valid(minimum_alive_secs:int) -> bool :
+    # If reached this point, minimum_alive_secs can not be NONE
+    if not isinstance(minimum_alive_secs, int):
+        return False
+
+    if  (isinstance(minimum_alive_secs, int) and
+            minimum_alive_secs > config.MAX_REQUESTED_ALIVE_SECS or minimum_alive_secs < 0):
+        return False
+    else:
+        return True
+
+
+def handler(event, context):
+    client_id = event.get('client_id')
+    if not client_id or client_id.strip() == "":
+        return {
+            "statusCode": 400,
+            "body": {"error": "client_id is a required field"}
+        }
+    if not is_client_id_valid(client_id):
+        return {
+            "statusCode": 400,
+            "body": {"error": "client_id must be alpha-numeric"}
+        }
+    # if user passed in a non-integer minimum_alive_secs, this line will error out
+    minimum_alive_secs=config.DEFAULT_TOKEN_MIN_ALIVE_SECS if event.get('minimum_alive_secs') is None \
```

Comment:
Shouldn't this be `config.MINIMUM_ALIVE_SECS` since we allow it to be overridden by an environment variable?